watertap/unit_models/zero_order/tests/test_secondary_treatment_wwtp_zo.py

```diff
@@ -0,0 +1,256 @@
+        assert (pytest.approx(89.617057, rel=1e-5) ==
+                value(model.fs.unit.properties_treated[0].conc_mass_comp["foo"]))
+        assert (pytest.approx(0.0018414, rel=1e-5) ==
+                value(model.fs.unit.properties_byproduct[0].flow_vol))
+        assert (pytest.approx(488.267142, rel=1e-5) ==
+                value(model.fs.unit.properties_byproduct[0].conc_mass_comp["viruses_enteric"]))
+        assert (pytest.approx(511.73285, rel=1e-5) ==
+                value(model.fs.unit.properties_byproduct[0].conc_mass_comp["tss"]))
+        assert (pytest.approx(4.344496876e-7, rel=1e-5) ==
+                value(model.fs.unit.properties_byproduct[0].conc_mass_comp["foo"]))
+        assert (pytest.approx(6.552, abs=1e-5) ==
+                value(model.fs.unit.electricity[0]))
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_conservation(self, model):
+        for j in model.fs.params.component_list:
+            assert 1e-6 >= abs(value(
+                model.fs.unit.inlet.flow_mass_comp[0, j] -
+                model.fs.unit.treated.flow_mass_comp[0, j] -
+                model.fs.unit.byproduct.flow_mass_comp[0, j]))
+
+    @pytest.mark.component
+    def test_report(self, model):
+        stream = StringIO()
+
+        model.fs.unit.report(ostream=stream)
+
+        output = """
+====================================================================================
+Unit : fs.unit                                                             Time: 0.0
+------------------------------------------------------------------------------------
+    Unit Performance
+
+    Variables: 
+
+    Key                              : Value   : Fixed : Bounds
+                  Electricity Demand :  6.5520 : False : (0, None)
+               Electricity Intensity : 0.14000 :  True : (None, None)
+                Solute Removal [foo] :  0.0000 :  True : (0, None)
+                Solute Removal [tss] : 0.94231 :  True : (0, None)
+    Solute Removal [viruses_enteric] : 0.89910 :  True : (0, None)
+                      Water Recovery :  1.0000 :  True : (1e-08, 1.0000001)
+
+------------------------------------------------------------------------------------
+    Stream Table
+                                         Inlet   Treated  Byproduct
+    Volumetric Flowrate                0.013000 0.011159  0.0018414
+    Mass Concentration H2O               769.23   896.17 4.3445e-07
+    Mass Concentration viruses_enteric   76.923   9.0424     488.27
+    Mass Concentration tss               76.923   5.1700     511.73
+    Mass Concentration foo               76.923   89.617 4.3445e-07
+====================================================================================
+"""
+
+        assert output in stream.getvalue()
+
+db = Database()
+params = db._get_technology("secondary_treatment_wwtp")
+
+
+class Test_SecondaryTreatmentWWTPZOsubtype:
```

Comment:
Why there is a subtype test here? The yaml file does not have a subtype.

Reply:
Oh, it will in the future, so the test is setup for that.